src/AmplNLWriter.jl

```diff
@@ -23,18 +23,31 @@ export AmplNLSolver,
     getsolvemessage,
     getsolveexitcode
 
+# Functionify the solver command so it can be called as
+# `solver_command() do path`.
+_solver_command(x::String) = f -> f(x)
```

Comment:
A (probable) silly question. What does `f` refer to here?

Reply:
Not a silly question. This method is very non-obvious. I've added a docstring.

Reply:
I get it.
It uses a function as the argument to run the executable located at `x::String`.

Reply:
Correct.

The JLL packages (Ipopt_jll, Bonmin_jll, Couenne_jll) all use this approach, so it makes sense to unify on that.

Ideally, using Bonmin etc will become as easy as 
```Julia
using JuMP, AmplNLWriter, Bonmin_jll
model = Model(() -> AmplNLWriter.Optimizer(Bonmin_jll.amplexe))
```

Although I may even try to merge https://github.com/jump-dev/AmplNLWriter.jl/pull/125 into MOI as the `FileFormats.NL` submodule, in which case you could go
```Julia
using JuMP, Bonmin_jll
model = Model(() -> MOI.FileFormats.NL.Optimizer(Bonmin_jll.amplexe))
```